src/hooks/useENSSearch.ts

```diff
@@ -42,9 +55,17 @@ export default function useENSSearch({
         valid: false,
       };
     }
-
-    const isAvailable = await getAvailable(name);
-    const rentPrice = await getRentPrice(name, duration);
+    const [
+      isAvailable,
+      rentPrice,
+      registrationDate,
+      nameExpires,
+    ] = await promiseUtils.PromiseAllWithFails([
```

Comment:
I believe this will also swallow errors – are we sure about that?

Reply:
i dont think it makes a difference from the user perspective but yeah looking back its probably not a good idea to blanket catch errors. i pushed a change that catches and rethrows the errors from the individual promises if applicable/relevant. 

if you think thats too messy looking, i could just break this up into two Promise.alls - one for availablility/rent price, the other for registration/expiration date (would only get called if name is unavailable).